scylla/src/transport/load_balancing/latency_aware.rs

```diff
@@ -251,3 +251,332 @@ impl<'a> Iterator for IteratorWithSkippedNodes<'a> {
+
+        assert_eq!(expected_plans, plans);
+    }
+
+    #[tokio::test]
+    async fn latency_aware_policy_does_not_penalise_if_exclusion_threshold_not_crossed() {
+        let policy = LatencyAwarePolicy::default();
+        let mut cluster = tests::mock_cluster_data_for_round_robin_and_latency_aware_tests();
+
+        let min_avg = Duration::from_millis(10);
+
+        tests::set_nodes_latency_stats(
+            &mut cluster,
+            &[
+                (
+                    1,
+                    Some(TimestampedAverage {
+                        timestamp: Instant::now(),
+                        average: Duration::from_secs_f64(
+                            policy.exclusion_threshold * 0.95 * min_avg.as_secs_f64(),
+                        ),
+                        num_measures: policy.minimum_measurements,
+                    }),
+                ),
+                (
+                    3,
+                    Some(TimestampedAverage {
+                        timestamp: Instant::now(),
+                        average: min_avg,
+                        num_measures: policy.minimum_measurements,
+                    }),
+                ),
+            ],
+        );
+
+        let plans = (0..16)
+            .map(|_| {
+                tests::get_plan_and_collect_node_identifiers(
+                    &policy,
+                    &tests::EMPTY_STATEMENT,
+                    &cluster,
+                )
+            })
+            .collect::<HashSet<_>>();
+
+        let expected_plans = vec![vec![1, 2, 3, 4, 5]]
+            .into_iter()
+            .collect::<HashSet<Vec<_>>>();
+
+        assert_eq!(expected_plans, plans);
+    }
+
+    #[tokio::test]
+    async fn latency_aware_policy_does_not_penalise_if_retry_period_expired() {
+        let policy =
+            LatencyAwarePolicy::new(2., Duration::from_millis(10), Duration::from_millis(10), 20);
+        let mut cluster = tests::mock_cluster_data_for_round_robin_and_latency_aware_tests();
+
+        let min_avg = Duration::from_millis(10);
+
+        tests::set_nodes_latency_stats(
+            &mut cluster,
+            &[
+                (
+                    1,
+                    Some(TimestampedAverage {
+                        timestamp: Instant::now(),
+                        average: Duration::from_secs_f64(
+                            policy.exclusion_threshold * 1.5 * min_avg.as_secs_f64(),
+                        ),
+                        num_measures: policy.minimum_measurements,
+                    }),
+                ),
+                (
+                    3,
+                    Some(TimestampedAverage {
+                        timestamp: Instant::now(),
+                        average: min_avg,
+                        num_measures: policy.minimum_measurements,
+                    }),
+                ),
+            ],
+        );
+
+        tokio::time::sleep(policy.retry_period).await;
```

Comment:
I'd recommend sleeping a bit more (e.g. twice the amount of milliseconds) than the retry period - the tokio timer's precision isn't perfect.

Reply:
On the other hand, retry period is by default measured in seconds, so its precision should be good enough. Should we bother with much smaller retry period configurations?